devel/xxdiff/Portfile

```diff
@@ -1,62 +1,100 @@
-                    differences between two or three files, or between \
-                    two directories, and can be used to produce a \
-                    merged version.
+                    differences between two or three files, or between two \
+                    directories, and can be used to produce a merged version. \
+                    It is inspired by the venerable SGI xdiff utility
+
 homepage            http://furius.ca/xxdiff/
 
-depends_lib         port:bison \
-                    port:flex \
-                    port:qt3 \
-                    port:libGLU
 
-# xxdiff links with libGLU if it is installed; so make sure it is
-# installed already.  See also: https://trac.macports.org/ticket/47693
+depends_lib         port:bison \
+                    port:flex
 
-master_sites        sourceforge:project/xxdiff/xxdiff/${version}/
-use_bzip2           yes
-checksums           md5     301ff399a1e4be8b68c38c3e27c0b1ad \
-                    sha1    ab2848fd9279f55b5db270f319ad4791154ecf82 \
-                    rmd160  12e7527f39713b925f05b687c2ad8b60459bb670
+fetch.type          hg
+hg.url              https://bitbucket.org/blais/xxdiff
+hg.tag              815d97a7a0a1
+distname            ${name}-5.0
+worksrcdir          ${distname}
 
-# https://trac.macports.org/ticket/41600#comment:47
-patchfiles          bison3.patch
+patchfiles-append   patch-xxdiffpro.diff
 
-# "Configure" and build adapted from README.build.
+set worksrcpath     ${workpath}/${worksrcdir}
+configure.pre_args  -f ${worksrcpath}/src/Makefile.bootstrap \
+                     MAKEDIR=${worksrcpath}/src
+configure.dir       ${workpath}/build
+configure.cmd       make
+configure.pre_args-append \
+                    QMAKE=${qt_qmake_cmd} QMAKEOPTS="-spec ${qt_qmake_spec}"
+build.dir           ${configure.dir}
 
-build.type          gnu
+# MacPorts base can add -stdlib=libc++ to configure.cxxflags, in which case it also has
+# to be added to configure.ldflags
+if {![string match "*stdc=libc++*" ${configure.cxxflags}]} {
+    configure.ldflags-append \
+                    -stdlib=libc++
+}
 
-configure.dir       ${worksrcpath}/src
-configure.cmd       ${build.cmd}
-configure.pre_args  -f Makefile.bootstrap
-configure.args      QTDIR='${prefix}' makefile
+post-patch {
+    reinplace "s|@@CFLAGS@@|${configure.cflags}|g" ${worksrcpath}/src/xxdiff.pro
+    reinplace "s|@@CXXFLAGS@@|${configure.cxxflags}|g" ${worksrcpath}/src/xxdiff.pro
+    reinplace "s|@@LFLAGS@@|${configure.ldflags}|g" ${worksrcpath}/src/xxdiff.pro
+    reinplace "s|@@QT_INCLUDES_DIR@@|${qt_includes_dir}|g" ${worksrcpath}/src/xxdiff.pro
+}
 
-build.dir           ${configure.dir}
-build.args          QTDIR='${prefix}'
+# xxdiff has a somewhat flaky way of using flex and bison. Make sure everything works:
+pre-configure {
+    file mkdir ${configure.dir}
+    # this seems to be necessary to ensure both parsers will be generated before their headerfiles are required
```

Comment:
Believe me I tried and gave up because to be honest I don't understand how it's supposed to work.
The target I remove has been introduced very recently, as part of a reintroduction of support for the platform bison and flex.

In case you have ideas how to fix this, the exact error is 
`../blais-xxdiff-815d97a7a0a1/src/resParser.cpp:40:10: fatal error: 'resParser_lex.h' file not found
`
When I follow the instructions in the README.build file and unpatched source, I get the following error:
```
WARNING: Failure to find: resParser_lex.h
WARNING: Failure to find: resParser_bison23.y
```

You're not wrong that I could report this upstream, but apparently it must work for them and since I've been using this workaround for a long time now it's no longer a big deal for me.